src/DataShare/DataShare/SynchronizationSettings/RemoveAzureDataShareSynchronizationSetting.cs

```diff
@@ -0,0 +1,162 @@
+
+namespace Microsoft.Azure.Commands.DataShare.SynchronizationSetting
+{
+    using System;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.DataShare.Common;
+    using Microsoft.Azure.Commands.DataShare.Helpers;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.DataShare;
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Properties;
+    using System.Globalization;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+
+    /// <summary>
+    /// Defines the Remove-AzDataShareSynchronizationSetting cmdlet.
+    /// </summary>
+    [Cmdlet(
+         "Remove",
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareSynchronizationSetting",
+         DefaultParameterSetName = ParameterSetNames.FieldsParameterSet,
+         SupportsShouldProcess = true),
+     OutputType(typeof(bool))]
+    public class RemoveAzureDataShareSynchronizationSetting : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "The resource group name of the azure data share account")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.FieldsParameterSet,
+            HelpMessage = "Azure Data Share Account name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// The name of the azure data share.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data share name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string ShareName { get; set; }
+
+        /// <summary>
+        /// The name of the synchronization setting to delete.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
You should not use ValueFromPipelineByPropertyName for any parameter other than 'ResourceId', in any cmdlet